services/pgnotifier/pgnotifier.go

```diff
@@ -219,6 +235,63 @@ func (notifier *PgNotifierT) trackBatch(batchID string, ch *chan []ResponseT) {
 	})
 }
 
+func (notifier *PgNotifierT) trackAsyncBatch(batchID string, ch *chan []ResponseT) {
+	rruntime.GoForWarehouse(func() {
+		for {
+			time.Sleep(trackBatchInterval)
+			// keep polling db for batch status
+			// or subscribe to triggers
+			pkgLogger.Info("Tracking the async batch that's uploaded")
+			stmt := fmt.Sprintf(`SELECT count(*) FROM %s WHERE batch_id='%s' AND status!='%s' AND status!='%s'`, queueName, batchID, SucceededState, AbortedState)
+			var count int
+			err := notifier.dbHandle.QueryRow(stmt).Scan(&count)
+			if err != nil {
+				pkgLogger.Errorf("PgNotifier: Failed to query for tracking jobs by batch_id: %s, connInfo: %s", stmt, notifier.URI)
+				panic(err)
+			}
+
+			if count == 0 {
+				stmt = fmt.Sprintf(`SELECT payload, status, error FROM %s WHERE batch_id = '%s'`, queueName, batchID)
+				rows, err := notifier.dbHandle.Query(stmt)
+				if err != nil {
+					panic(err)
```

Comment:
panic would halt the program as a whole right. Let's capture the error on this and then move on. Wdyt ?

Reply:
Yep. Will capture the error instead of panic. I had trouble while tesitng as well.